### PR TITLE
# 215 reservation logic ListView and Test 215rv5

### PR DESCRIPTION
**Pull request by ebercamargo**

### CODE REVIEW

**Comment by 0crat** [comment]
Job [#463](https://github.com/timelesslounge/timelessis/issues/463) is now in scope, role is `REV`

**Comment by 0crat** [comment]
This pull request [#463](https://github.com/timelesslounge/timelessis/issues/463) is assigned to @vryazanov[/z](https://www.0crat.com/u/vryazanov), here is [why](https://www.0crat.com/footprint/CF9LG1YCA/fcbbeb58-3d39-4af6-ae5b-9612947cd9c0); the budget is 15 minutes, see [§4](http://www.zerocracy.com/policy.html#4); please, read [§27](http://www.zerocracy.com/policy.html#27) and when you decide to accept the changes, inform @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa) (the architect) right in this ticket; if you decide that this PR should not be accepted ever, also inform the architect; this [blog post](http://www.yegor256.com/2015/02/09/serious-code-reviewer.html) will help you understand what is expected from a code reviewer; there will be a monetary reward for this job

**Comment by ebercamargo** [comment]
timeless/reservations/views.py - Arrumei o View com gerenic
tests/test_reservation.py - I placed the validation with login, as informed.

File: tests/test_reservation.py [hunk 1]
```diff
@@ -1,8 +1,9 @@
 from datetime import datetime
 from http import HTTPStatus
 
+import  pytest
 
+@pytest.mark.skip(reason="authentication not implemented for reservations")
```
Comment (vryazanov) [inline tests/test_reservation.py]: @ebercamargo have a look at https://github.com/timelesslounge/timelessis/blob/master/tests/integration/it_companies_test.py#L21

File: timeless/reservations/views.py [hunk 2]
```diff
@@ -92,45 +100,55 @@ def get(self, company_id):
         return jsonify(reservations_json)
 
 
-@bp.route("/")
-def list_reservations(reservations):
+@bp.route("/create", methods=("GET", "POST"))
+def create():
```
Comment (vryazanov) [inline timeless/reservations/views.py]: @ebercamargo we have `CreateView` for this operation

File: timeless/reservations/views.py [hunk 3]
```diff
@@ -92,45 +100,55 @@ def get(self, company_id):
...
+        "reservations/create_edit.html", action="edit",
         id=id
     )
 
 
-@bp.route("/delete", methods=["POST"])
-def delete():
-    flash("Delete not yet implemented")
+@bp.route("/delete/<int:id>", methods=["POST"])
+def delete(id):
```
Comment (vryazanov) [inline timeless/reservations/views.py]: @ebercamargo `DeleteView` should be there (see `timeless/views.py`)

Comment (ebercamargo) [reply timeless/reservations/views.py]: Hi, @vryazanov The issue that was opened was to create only one View. Doesn't that other issue?

File: tests/test_reservation.py (hunk 1 above)
Comment (ebercamargo) [reply tests/test_reservation.py]: @vryazanov  changed as requested.

File: timeless/reservations/views.py (hunk 3 above)
Comment (vryazanov) [reply timeless/reservations/views.py]: @ebercamargo why did you touch this code if it's another issue :)
current implementation that you brought is wrong

Comment (ebercamargo) [reply timeless/reservations/views.py]: @vryazanov the change I made does not satisfy?

Comment (ebercamargo) [reply timeless/reservations/views.py]: I downloaded the repository and changed it. My code is just this:

`class ReservationsListView(views.ListView):
    """ List the Items """
    model = Reservation
    template_name = "reservations/list.html"

ReservationsListView.register(bp, "/")`

Comment (vryazanov) [reply timeless/reservations/views.py]: @ebercamargo hm, looks like github shows incorrect diff. pls fix merge conflics and then I'll have a look one more time

**Comment by 0crat** [comment]
@ypshenychka[/z](https://www.0crat.com/u/ypshenychka) please review this job completed by @vryazanov[/z](https://www.0crat.com/u/vryazanov), as in [§30](http://www.zerocracy.com/policy.html#30); the job will be fully closed and all payments will be made when the quality review is completed

**Comment by 0crat** [comment]
The job [#463](https://github.com/timelesslounge/timelessis/issues/463) is now out of scope

**Comment by 0crat** [comment]
Payment to `ARC` for a closed pull request, as in [§28](http://www.zerocracy.com/policy.html#28): +10 point(s) just awarded to @emilianodellacasa[/z](https://www.0crat.com/u/emilianodellacasa)

**Comment by ypshenychka** [comment]
@ebercamargo According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>Messages in a job [always](http://www.yegor256.com/2014/11/24/principles-of-bug-tracking.html#4-avoid-noiseaddress-your-comments) start with a name of a user they are addressed to.

Please correct your code review comments and messages by indicating an addressee in the beginning.

**Comment by ypshenychka** [comment]
@ebercamargo According to our [QA Rules](http://www.zerocracy.com/policy.html#42):
>Pull request description explains the solution proposed and contains a link to the original ticket it is related to

Please add appropriate description.

**Comment by ypshenychka** [comment]
@ebercamargo ping. Respond to my remarks, please.

**Comment by ebercamargo** [comment]
> @ebercamargo ping. Respond to my remarks, please.

@ypshenychka Information adjusted, please check if everything is ok, tks.

**Comment by ypshenychka** [comment]
@0crat quality acceptable

**Comment by 0crat** [comment]
Order was finished, quality is "acceptable": +15 point(s) just awarded to @vryazanov[/z](https://www.0crat.com/u/vryazanov)

**Comment by 0crat** [comment]
Quality review completed: +6 point(s) just awarded to @ypshenychka[/z](https://www.0crat.com/u/ypshenychka)

**Pull request closed without merging**